update gitignore and improve project organization exercise

vinisalazar opened the pull request:
Hi,

This is related to #55.

Summary of changes:
  - Renames challenge from 'Exercise' to 'Creating project files'.
  - Adds 'mkdir results' command.
  - Improves explanation of .gitignore.

Best,
Vini

vinisalazar committed:
_episodes/02-project-setup.md: Improving challenge.
  - Related to #55.
  - Renames challenge from 'Exercise' to 'Creating project files'.
  - Adds 'mkdir results' command.
  - Improves explanation of .gitignore.

brownsarahm commented on _episodes/02-project-setup.md:
```diff
@@ -110,6 +111,7 @@ replacing the `__` on each line
 > touch __/reproduce_paper.py # code to re-run the analyses reported in your methods paper about the package
 > touch __/helper_functions.py # auxilliary functions for routine tasks associated with the novel method
 > touch __/how_to_setup.md # details to help others prepare equivalent experiments to those presented in your paper
+> touch __/.gitignore # flags files that should be ignored by version control, such as big data files
```
I don't like adding this to the exercise. this becomes a question about how git works.

brownsarahm commented on _episodes/02-project-setup.md:
```diff
@@ -122,8 +124,18 @@ replacing the `__` on each line
+> > {: .language-none}
+> > Which would prevent `data/raw_data.csv` and any files in the `results/` directory from being included in version
+> > control. This is fairly common when working with very large files, and can be addressed by adding instructions on
+> > how to download the data.
```
Lets put all of the talk about gitignore in its own section.

brownsarahm commented:
This doesn't really address #55 though; this improves gitignore, but doesn't add advice about how to work with data and results very much.

vinisalazar committed:
_episodes/02-project-setup.md: Improving .gitignore and results explanation.
  - Related to #55.
  - Code review for #81.
  - Add callout block for 'Where to store results?'
  - Add section of configuration files and .gitignore.

vinisalazar committed:
_episodes/02-project-setup.md: update branch.
  - Code review for #81.
